### PR TITLE
HPCC-28209 Fix Process Down warning for thor slaves in Target Cluster Preflight page

### PR DESCRIPTION
**Pull request by wangkx**
This PR fixes 2 issues in the preflight function for a bare-metal
environment. 
1. Incorrect pid file names are used in the existing preflight function 
to check Thor slave processes and Thor spare processes. In this fix, 
the node sequence numbers in the nodeGroup are used to create 
the pid file names for checking those processes.
2. The existing code logs a 'Cannot find node group ...' message when
it cannot find the node group for a Thor spare process, even if the
Thor cluster does not have a spare process. This issue is fixed.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/services/ws_machine/ws_machineService.cpp [hunk 1]
```diff
@@ -850,10 +847,11 @@ void Cws_machineEx::getThorProcesses(IConstEnvironment* constEnv, IPropertyTree*
             continue;
         }
 
-        //Each thor slave is a process. The i is used to check whether the process is running or not.
-        for (unsigned i = 1; i <= slavesPerNode; i++)
-            setProcessRequest(machineInfoData, uniqueProcesses, netAddress.str(), addressRead.str(),
-                processType, processName, directory, i);
+        //Each thor slave is a process. Each process has a PID file.
+        //The PID file is useed to get the information about the process.
```
Comment (jakesmith) [inline esp/services/ws_machine/ws_machineService.cpp:851]: typo: useed -> used

**Review by jakesmith — COMMENTED** [review]
@wangkx - looks good. Please fix typo and squash.

**Comment by wangkx** [comment]
@jakesmith the typo is fixed. Please review.

**Review by jakesmith — APPROVED** [review]
@wangkx - looks good.

**Comment by wangkx** [comment]
@ghalliday Please review and merge.

**Comment by ghalliday** [comment]
@wangkx the change looks fine I don't understand the commit message though.  Please can you reword it so it makes sense in a change log?

**Comment by wangkx** [comment]
@ghalliday please review the commit message.

**Comment by ghalliday** [comment]
The message:
```
HPCC-28209 Use node sequence number to Thor slave process
```
Still doesn't make sense on its own as a summary line.

Possibly
HPCC-28209 Use the correct name when retrieving the thor slave log file in cluster summary
or
HPCC-28209 Fix problem retrieving the thor slave log file in target cluster summary

Although I suspect neither of those correctly reflect the symptom the user would have seen.

**Comment by wangkx** [comment]
How about: HPCC-28209 Fix Process Down warning for thor slaves in Target Cluster Preflight page ? That is the symptom the user would have seen when incorrect correct name is used.

**Comment by ghalliday** [comment]
That sounds good to me,  much more helpful.  Thanks

**Comment by ghalliday** [comment]
@wangkx please can you change the commit message as well as the title.

**Comment by wangkx** [comment]
Will do

**Comment by wangkx** [comment]
@ghalliday I just changed both the commit message and the title.

**Pull request merged**